core/src/main/java/org/infinispan/persistence/sifs/NonBlockingSoftIndexFileStore.java

```diff
@@ -510,7 +510,11 @@ protected boolean isIndexLoaded() {
    @Override
    public CompletionStage<Long> approximateSize(IntSet segments) {
       // Approximation doesn't pause the appender so the index can be slightly out of sync
-      return CompletableFuture.completedFuture(index.approximateSize());
+      long totalSize = index.approximateSize();
+      IntSet matchingSegments = IntSets.mutableCopyFrom(segments);
+      matchingSegments.retainAll(matchingSegments);
```

Comment:
You mean `matchingSegment.retainAll(temporaryTable.getOwnedSegments()) ?

Reply:
Oops!

Reply:
Fixed